src/actions/MatrixActionCreators.js

```diff
@@ -54,14 +52,33 @@ function createAccountDataAction(matrixClient, accountDataEvent) {
     };
 }
 
+/**
+ * This object is responsible for dispatching actions when certain events are emitted by
+ * the given MatrixClient.
+ */
 export default {
+    // A list of callbacks to call to unregister all listeners added
     _matrixClientListenersStop: [],
 
+    /**
+     * Start listening to certain events from the MatrixClient and dispatch actions when
+     * they are emitted.
+     * @param {MatrixClient} matrixClient the MatrixClient to listen to events from
+     */
     start(matrixClient) {
         this._addMatrixClientListener(matrixClient, 'sync', createSyncAction);
         this._addMatrixClientListener(matrixClient, 'accountData', createAccountDataAction);
     },
 
+    /**
+     * Start listening to events emitted by matrixClient, dispatch an action created by the
+     * actionCreator function.
+     * @param {MatrixClient} matrixClient a MatrixClient to register a listener with.
+     * @param {string} eventName the event to listen to on MatrixClient.
+     * @param {function} actionCreator a function that should return an action to dispatch
```

Comment:
could you also specify that the function will be passed the MatrixClient event?

Reply:
I'm assuming you meant MatrixClient itself

Reply:
both, but yes.